llvm/lib/Target/AMDGPU/BUFInstructions.td

```diff
@@ -1735,8 +1737,11 @@ defm : SIBufferAtomicPat<"SIbuffer_atomic_dec", i64, "BUFFER_ATOMIC_DEC_X2">;
 let OtherPredicates = [HasAtomicCSubNoRtnInsts] in
 defm : SIBufferAtomicPat<"SIbuffer_atomic_csub", i32, "BUFFER_ATOMIC_CSUB", ["noret"]>;
 
-let SubtargetPredicate = isGFX12Plus in {
+let SubtargetPredicate = HasAtomicBufferPkAddBF16Inst in {
   defm : SIBufferAtomicPat_Common<"SIbuffer_atomic_fadd", v2bf16, "BUFFER_ATOMIC_PK_ADD_BF16_VBUFFER">;
```

Comment:
Should it use OtherPredicates = [HasAtomicBufferPkAddBF16Inst] and SubtargetPredicate = isGFX12Plus because VBUFFER opcode is used?

Reply:
I think these should be inverted, the inner predicate should be the buffer thing

Reply:
I'm not really sure what VBUFFER is, but as far as I can tell the predicates are broken anyway. If you do select the non-VBUFFER version, something else breaks later and we don't have the negated predicate on the not-VBUFFER patterns

Reply:
VBUFFER is a new encoding in GFX12 which replaces the old MTBUF and MUBUF encodings. We have different pseudos for VBUFFER (which should only be selected on GFX12+) and MTBUF/MUBUF (which should only be selected pre-GFX12).